src/goInstallTools.ts

```diff
@@ -484,24 +516,53 @@ function getMissingTools(goVersion: SemVersion): Promise<string[]> {
 	});
 }
 
-// If langserver needs to be used, but is not installed, this will prompt user to install and Reload
-// If langserver needs to be used, and is installed, this will return true
-// Returns false in all other cases
-export function checkLanguageServer(): boolean {
+/**
+ * Gets the absolute path to the language server to be used.
+ * If the required tool is not available, then user is prompted to install it.
+ * This supports the language servers from both Google and Sourcegraph with the
+ * former getting a precedence over the latter
+ */
+export function getLanguageServerToolPath(): string {
 	const latestGoConfig = vscode.workspace.getConfiguration('go');
```

Comment:
nit: since we want to enable undefined to not be part of the type domains , we might want to have the signature string | undefined

Reply:
Done